folly/Traits.h

```diff
@@ -639,6 +639,15 @@ struct IsRelocatable<std::pair<T, U>>
 template <typename T, typename... Ts>
 using IsOneOf = StrictDisjunction<std::is_same<T, Ts>...>;
 
+template <typename, template <typename...> class>
+struct is_instantiation_of  : std::false_type {};
+
+template <typename... Ts, template <typename...> class C>
+struct is_instantiation_of<C<Ts...>, C> : std::true_type {};
```

Comment:
Looking at `std::is_base_of`, params are backwards.

Reply:
We can swap the order, no biggie. What do you think of the idea itself?

Reply:
Well, not the biggest fan because:
* It can't be sufficiently generic and work with non-type template parameters.
* It doesn't work through aliases: https://godbolt.org/z/nzkt-S.

But there have been repeat grumblings about not having it, so maybe it's worth it anyway.

Given the limitations, I would personally prefer it to be in `namespace folly::detail`.

Reply:
Right, I find that the ability to not work through aliases is a bigger deal than the mixing of types v.s. non-types. With the mixing, you can always just stay in type space and wrap the values when you have control over them.

In any case, I moved it to a detail namespace. I suspect this is a useful enough utility (despite its limitations we've pointed out) to have in the folly toolbox for other libs/apps to use that already depend on folly.

Reply:
@yfeldblum by the way, despite not being ideal, it shouldn't be surprising that this doesn't work with alias templates. This comes from the fact that you can't pass an alias template as a template template parameter.